serializers/src/glTF/2.0/Extensions/KHR_materials_clearcoat.ts

```diff
@@ -0,0 +1,102 @@
+import { BaseTexture } from 'babylonjs/Materials/Textures/baseTexture';
+import { IKHRMaterialsClearcoat } from 'babylonjs-gltf2interface';
+import { PBRMetallicRoughnessMaterial } from "babylonjs/Materials/PBR/pbrMetallicRoughnessMaterial";
+
+const NAME = "KHR_materials_clearcoat";
+
+/**
+ * @hidden
+ */
+export class KHR_materials_clearcoat implements IGLTFExporterExtensionV2 {
+    /** Name of this extension */
+    public readonly name = NAME;
+
+    /** Defines whether this extension is enabled */
+    public enabled = true;
+
+    /** Defines whether this extension is required */
+    public required = false;
+
+    private _exporter: _Exporter;
+
+    private _wasUsed = false;
+
+    constructor(exporter: _Exporter) {
+        this._exporter = exporter;
+    }
+
+    public dispose() {
+    }
+
+    /** @hidden */
+    public get wasUsed() {
+        return this._wasUsed;
+    }
+
+    public postExportMaterialAdditionalTextures?(context: string, node: IMaterial, babylonMaterial: Material): BaseTexture[] {
+        let additionalTextures: BaseTexture[] = [];
+        if (babylonMaterial instanceof PBRMaterial) {
+            if (babylonMaterial.clearCoat.isEnabled) {
+                if (babylonMaterial.clearCoat.texture) {
+                    additionalTextures.push(babylonMaterial.clearCoat.texture);
+                }
+                if (!babylonMaterial.clearCoat.useRoughnessFromMainTexture && babylonMaterial.clearCoat.textureRoughness) {
+                    additionalTextures.push(babylonMaterial.clearCoat.textureRoughness);
+                }
+                if (babylonMaterial.clearCoat.bumpTexture) {
+                    additionalTextures.push(babylonMaterial.clearCoat.bumpTexture);
+                }
+                return additionalTextures;
+            }
+        }
+
+        return [];
+    }
+
+    public postExportMaterialAsync?(context: string, node: IMaterial, babylonMaterial: Material): Promise<IMaterial> {
+        return new Promise((resolve, reject) => {
+            if (babylonMaterial instanceof PBRBaseMaterial) {
+                if (!babylonMaterial.clearCoat.isEnabled) {
+                    resolve(node);
+                    return;
+                }
+
+                this._wasUsed = true;
+
+                if (node.extensions == null) {
+                    node.extensions = {};
+                }
+
+                let clearCoatTextureInfo = this._exporter._glTFMaterialExporter._getTextureInfo(babylonMaterial.clearCoat.texture);
+                let clearCoatTextureRoughnessInfo;
+                if (babylonMaterial.clearCoat.useRoughnessFromMainTexture) {
+                    clearCoatTextureRoughnessInfo = this._exporter._glTFMaterialExporter._getTextureInfo((babylonMaterial as PBRMaterial).metallicTexture
+                                                                                                        ?? (babylonMaterial as PBRMetallicRoughnessMaterial).metallicRoughnessTexture);

```

Comment:
This is unfortunate. I wonder if we can expose `_metallicTexture` somehow instead of having to switch. The two properties both point to `_metallicTexture`.

Reply:
I suppose, but this feels right as an external consumer of these materials

Reply:
Doesn't feel right to me. We shouldn't have to cast to the leaf type in order to access common properties. If another child material were added that remaps `_metallicTexture` to something else, we will have to add it here.

Reply:
Exposed the PBRBaseMaterial properties.